framework/src/meshgenerators/ElementDeletionGeneratorBase.C

```diff
@@ -52,6 +52,15 @@ ElementDeletionGeneratorBase::generate()
       deleteable_elems.insert(elem);
   }
 
+  // Set to be merged into deleteable elems, used to avoid derefencing a dangling pointer
+  std::set<Elem *> deleteable_parents;
+
+  // check for dangling interior parents
+  for (auto & elem : mesh->element_ptr_range())
```

Comment:
this is costing you another loop on all the mesh

Reply:
ok so check the element dimension before doing a search. this will reduce the complexity from:
N_mesh * N_elem_deleted to
N_lower-d-mesh * N_elem_deleted

and that should be enough